history/storage.py

```diff
@@ -28,6 +28,41 @@
 logger = logging.getLogger('{}:'.format(__name__))
 
 
+def _try_decoding_response_body(response_body, encoding):
+    # Try to guess encoding
+    try:
+        return response_body.decode(encoding)
+    except Exception:
+        pass
+
+    try:
+        encoding = 'utf-8'
+        return encoding, response_body.decode(encoding)
+    except Exception:
+        pass
+
+    try:
+        encoding = 'latin-1'
+        return encoding, response_body.decode(encoding)
+    except Exception:
+        pass
+
+    try:
+        encoding = 'ISO-8859-1'
+        return encoding, response_body.decode(encoding)
+    except Exception:
+        pass
+
+    try:
+        encoding = 'utf-8'
+        return encoding, response_body.decode(encoding, 'ignore')
```

Comment:
May be for this one we could return something like `utf-8/ignore` to properly distinguish from strict UTF-8 decoding.